File: app/src/main/java/se/gustavkarlsson/skylight/android/gui/activities/main/fragments/aurora_chance/LocationPresenter.java
package se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance;

import android.location.Address;
import android.widget.TextView;

import static android.view.View.INVISIBLE;
import static android.view.View.VISIBLE;

public class LocationPresenter {
	private final TextView locationTextView;

	public LocationPresenter(TextView locationTextView) {
		this.locationTextView = locationTextView;
	}

	void update(Address address) {
		if (address == null) {
			locationTextView.setVisibility(INVISIBLE);
		} else {
			locationTextView.setVisibility(VISIBLE);
			String locationString = address.getLocality();
			locationTextView.setText(locationString);
		}
	}
}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/gui/activities/main/fragments/aurora_chance/LocationPresenter.kt
package se.gustavkarlsson.skylight.android.gui.activities.main.fragments.aurora_chance

import android.view.View.INVISIBLE
import android.view.View.VISIBLE
import android.widget.TextView
import kotlinx.coroutines.experimental.android.UI
import kotlinx.coroutines.experimental.async
import se.gustavkarlsson.skylight.android.services.Presenter

class LocationPresenter(
	private val locationTextView: TextView
) : Presenter<String?> {

    override fun present(value: String?) {
		async(UI) {
			if (value == null) {
				locationTextView.visibility = INVISIBLE
			} else {
				locationTextView.text = value
				locationTextView.visibility = VISIBLE
			}
		}
    }
}
